src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentProperty.java

```diff
@@ -99,7 +100,13 @@ public String getFieldName() {
 	// DATACOUCH-145: allows SDK's @Id annotation to be used
 	@Override
 	public boolean isIdProperty() {
-		return isAnnotationPresent(Id.class) || super.isIdProperty()
-				|| this.getFieldName().toLowerCase(Locale.ROOT).equals("id");
+		if (isAnnotationPresent(Id.class) || super.isIdProperty()){
+			return true;
+		}
+		// is field named "id"
+		if(getField() != null && this.getFieldName().toLowerCase(Locale.ROOT).equals("id")){
```

Comment:
In MongoDB, we simplify that check to:

```
SUPPORTED_ID_PROPERTY_NAMES.contains(getName()) && !hasExplicitFieldName()
```

Here, you could check for e.g. `"id".equals(getName())`.

Reply:
Done. Thanks!